modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PkmstServerCodegen.java

```diff
@@ -0,0 +1,1457 @@
+
+        // Only bother with reconciliation if the parent model has enums.
+        if (!parentCodegenModel.hasEnums) {
+            return codegenModel;
+        }
+
+        // Get the properties for the parent and child models
+        final List<CodegenProperty> parentModelCodegenProperties = parentCodegenModel.vars;
+        List<CodegenProperty> codegenProperties = codegenModel.vars;
+
+        // Iterate over all of the parent model properties
+        boolean removedChildEnum = false;
+        for (CodegenProperty parentModelCodegenPropery : parentModelCodegenProperties) {
+            // Look for enums
+            if (parentModelCodegenPropery.isEnum) {
+                // Now that we have found an enum in the parent class,
+                // and search the child class for the same enum.
+                Iterator<CodegenProperty> iterator = codegenProperties.iterator();
+                while (iterator.hasNext()) {
+                    CodegenProperty codegenProperty = iterator.next();
+                    if (codegenProperty.isEnum && codegenProperty.equals(parentModelCodegenPropery)) {
+                        // We found an enum in the child class that is
+                        // a duplicate of the one in the parent, so remove it.
+                        iterator.remove();
+                        removedChildEnum = true;
+                    }
+                }
+            }
+        }
+
+        if (removedChildEnum) {
+            // If we removed an entry from this model's vars, we need to ensure hasMore is
+            // updated
+            int count = 0, numVars = codegenProperties.size();
+            for (CodegenProperty codegenProperty : codegenProperties) {
+                count += 1;
+                codegenProperty.hasMore = (count < numVars) ? true : false;
+            }
+            codegenModel.vars = codegenProperties;
+        }
+        return codegenModel;
+    }
+
+    private static String getAccept(Operation operation) {
+        String accepts = null;
+        String defaultContentType = "application/json";
+        if (operation.getProduces() != null && !operation.getProduces().isEmpty()) {
+            StringBuilder sb = new StringBuilder();
+            for (String produces : operation.getProduces()) {
+                if (defaultContentType.equalsIgnoreCase(produces)) {
+                    accepts = defaultContentType;
+                    break;
+                } else {
+                    if (sb.length() > 0) {
+                        sb.append(",");
+                    }
+                    sb.append(produces);
+                }
+            }
+            if (accepts == null) {
+                accepts = sb.toString();
+            }
+        } else {
+            accepts = defaultContentType;
+        }
+
+        return accepts;
+    }
+
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    public String getName() {
+        return "pkmst";
```

Comment:
What about naming it as "java-pkmst" to follow the convention "{lang}-{framework}"?

Reply:
hi @wing328  We have changed the server codegen java file accordingly as the convention suggest from pkmstservercodegen to JavaPKMSTServerCodegen. Above name "pkmst" is what the user sees. Please let us know if its fine.

Reply:
also added the missing typeInfoAnnotation.mustache file. Could you send the swagger yaml file used for testing.

Reply:
hi William we have refactored and merged the code.  Can you share some swagger yaml apart from pet store that we can use for testing. Also let us know if you have any queries. Our testing is in progress.
@wing328